tests/common/runner.rs

```diff
@@ -50,30 +53,100 @@ pub fn test_execute_revm<S: Storage + Clone + Send + Sync>(
 ) {
     let concurrency_level = thread::available_parallelism().unwrap_or(NonZeroUsize::MIN);
     assert_execution_result(
-        pevm::execute_revm_sequential(storage.clone(), spec_id, block_env.clone(), txs.clone()),
-        pevm::execute_revm(storage, spec_id, block_env, txs, concurrency_level),
+        &pevm::execute_revm_sequential(storage.clone(), spec_id, block_env.clone(), txs.clone()),
+        &pevm::execute_revm(storage, spec_id, block_env, txs, concurrency_level),
         false, // TODO: Parameterize this
     );
 }
 
+fn calculate_receipt_root(receipt_envelopes: Vec<ReceiptEnvelope>) -> B256 {
+    // create a BTreeMap (order must be ascending)
+    let entries = receipt_envelopes
+        .into_iter()
+        .enumerate()
+        .map(|(index, receipt)| {
+            let key_buffer = alloy_rlp::encode_fixed_size(&index);
+            let mut value_buffer = Vec::new();
+            receipt.encode_2718(&mut value_buffer);
+            (key_buffer, value_buffer)
+        })
+        .collect::<BTreeMap<_, _>>();
+
+    // calculate the MPT root
+    let mut hash_builder = alloy_trie::HashBuilder::default();
+    for (k, v) in entries {
+        hash_builder.add_leaf(alloy_trie::Nibbles::unpack(&k), &v);
+    }
+    hash_builder.root()
+}
+
+fn get_receipt_envelopes(
```

Comment:
Minor but `build` is better than `get` here. Throughout Rust's standard library `get` functions usually return references to existing data -- truly the act of getting, not allocating and returning ownership to new data!